core/src/main/scala/org/apache/spark/storage/BlockManager.scala

```diff
@@ -1285,6 +1314,9 @@ private[spark] class BlockManager(
 
     require(blockId != null, "BlockId is null")
     require(level != null && level.isValid, "StorageLevel is null or invalid")
+    if (isDecommissioning()) {
+      throw new BlockSavedOnDecommissionedBlockManagerException(blockId)
```

Comment:
why are we throwing exception here? doPut on a decommissioning executor might haven been called by the running task - so can this line lead to task failure?